library/std/src/net/ip.rs

```diff
@@ -958,18 +953,18 @@ impl PartialOrd<IpAddr> for Ipv4Addr {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl Ord for Ipv4Addr {
     fn cmp(&self, other: &Ipv4Addr) -> Ordering {
-        u32::from_be(self.inner.s_addr).cmp(&u32::from_be(other.inner.s_addr))
+        self.inner.to_be().cmp(&other.inner.to_be())
```

Comment:
The former `u32::from_be` meant that these were comparing on native-endian values.